### ISSUE TITLE
Hide files and folders in WSL starting with a "." in file explorer

### ISSUE THREAD

**Issue by artorias111** [microsoft/WSL#6460]
As explained in the title, it isn't any major issue but would be a neat addition if files and folders starting with a "." could be hidden when WSL files are viewed in the file explorer.

**Comment by benyaminl** [comment]
This's a neat feature tbh.
Waiting it to be implemented on WSL 9P

**Comment by Stanzilla** [comment]
That's more like an anti-feature to me, every dev will probably want to see dotfiles.

**Comment by benyaminl** [comment]
> 
> 
> That's more like an anti-feature to me, every dev will probably want to see dotfiles.

How about show hidden files, It should be the win win solution for it. As it's normal on Linux

**Comment by Biswa96** [comment]
WSL P9 mount shows dot files by default.

**Comment by benyaminl** [comment]
Yeah I got it, I mean this is a feature that is great to hide . files, and show file when enable hidden files.

**Comment by Biswa96** [comment]
Why do you want to hide a file or folder? Many programs save configuration in dot files.

**Comment by benyaminl** [comment]
No, nothing, it's just as the OP/TS stated, I just think it's a good idea to start with, because with `ls` we can't see dot files, and that's normal on *NIX right. Also on normal file explorer on any distro did have this feature baked in.

**Comment by slier81** [comment]
this should be a feature? why want to hide dot files? ever used linux? without the ability to hide dot files, it become so cluttered in the file manager.. i dont want to see dot files all the time.. if i want to see it, i can easily see it using terminal

**Comment by cdsantiago** [comment]
yes please, I want to be able to hide .files  in windows explorer

**Comment by nbx0** [comment]
+1

**Comment by funder7** [comment]
> That's more like an anti-feature to me, every dev will probably want to see dotfiles.

Not in windows explorer! 
At least, not always

**Comment by clchen-arcadia** [comment]
Yes, I would really like to see this. And presumably you would be able to use the checkbox to toggle whether or not to show these hidden files.

**Comment by funder7** [comment]
I think that there are many ways to approach this, basically the "issue" comes from how hidden files are managed by Windows & "Linux" filesystems: 
- the first stores the hidden attribute in each [file's metadata](https://learn.microsoft.com/en-us/windows/win32/fileio/file-attribute-constants), 
- the latter sees any file with a leading dot filename, as hidden.

My personal opinion, not knowing the actual architecture, is to add an adapter in a layer close to the Windows kernel (if not inside he kernel), in order to make the change transparent to any software, which will see the file as a regular Windows file with it's attributes. 
The other path is to edit every single software interacting with the filesystem, to address this difference, leaving to each developer/software house the implementation...

Ideally, I would do the first without any doubt...keeping an eye on file watchers: this logic should not be heavy, or it may slow down I/O operations when applied to large repos.

Cheers

**Comment by Wakeelfahmed** [comment]
A needed feature (to hide file/folders & then toggle to see them), I will also like the ability to change the icon of folders

**Comment by jusmccar** [comment]
Any update on this? It would be very nice to have.

**Comment by Tyrben** [comment]
Or provide an accessor to the Windows Metadata to hide the file from WSL.

```
wslbridge_set_file_metadata .my_file.txt "Hidden=true"
```
